src/Pretzel/Program.cs

```diff
@@ -1,89 +1,166 @@
+            },
+            new Option(new[] { "--source", "-s" }, "The path to the source site (default current directory)")
+            {
+                Argument = new Argument<string>(() => fileSystem.Directory.GetCurrentDirectory())
+            }
+        };
+#pragma warning restore S2223 // Non-constant static fields should not be visible
 
-        [Export("SourcePath")]
-        public string SourcePath { get; }
+        [Import]
+        public CommandCollection CommandCollection { get; set; }
 
-        private BaseParameters parameters { get; }
+        [Export]
+        public IFileSystem FileSystem { get; set; } = fileSystem;
 
-        public Program()
+        private static async Task<int> Main(string[] args)
         {
-            parameters = BaseParameters.Parse(Args, FileSystem);
-            SourcePath = parameters.Path;
-        }
+            Tracing.SetTrace(ConsoleTrace.Write);
 
-        private static string[] Args;
-        private static void Main(string[] args)
-        {
-            Args = args;
-            var program = new Program();
+            args = PatchSourcePath(args);
 
-            program.InitializeTrace();
-            Tracing.Info("starting pretzel...");
-            Tracing.Debug("V{0}", Assembly.GetExecutingAssembly().GetName().Version);
+            var parseResult = ParseArguments(args);
+            if (parseResult.hasError)
+                return -1;
 
-            using (program.Compose())
+            try
             {
+                InitializeTrace(parseResult.debug);
+                Tracing.Info("starting pretzel...");
+                Tracing.Debug("V{0}", Assembly.GetExecutingAssembly().GetName().Version);
 
-                if (program.parameters.Help || !args.Any())
+                using (var host = Compose(parseResult.debug, parseResult.safe, parseResult.source))
                 {
-                    program.ShowHelp();
-                    return;
+                    var program = new Program();
+                    host.SatisfyImports(program);
+                    var result = await program.Run(GlobalOptions, args);
+                    WaitForClose();
+                    return result;
                 }
-
-                program.Run();
+            }
+            catch (Exception ex)
+            {
+                Tracing.Error(ex.Message);
+                WaitForClose();
+                return -1;
             }
         }
 
-        private void InitializeTrace()
+        internal static (bool debug, bool safe, string source, bool hasError) ParseArguments(string[] args)
         {
-            Tracing.SetTrace(ConsoleTrace.Write);
+            var command = new RootCommand()
+            {
+                TreatUnmatchedTokensAsErrors = false
+            };
 
-            if (parameters.Debug)
+            foreach (var option in GlobalOptions)
+                command.AddOption(option);
+
+            var parseResult = new Parser(command).Parse(args);
```

Comment:
You could pass directly the options in `Parser`.

Reply:
Nope, that's not possible, cause it will tread unrecognized options/commands as errors

Reply:
Ah, I forgot that, sorry.